### PR TITLE
*tp9586 - Added types, fixed bug when checking props, fixed package

### PR DESCRIPTION
**Pull request by luxalpa**
* added typings for Typescript 
* fixed bug with react prop-types to allow for false returns
* fixed the package.json to use whitelisting for files instead of npmignore (see: https://medium.com/@jdxcode/for-the-love-of-god-dont-use-npmignore-f93c08909d8d)

### CODE REVIEW

File: src/Laika.js [hunk 1]
```diff
@@ -47,6 +47,6 @@ Laika.propTypes = {
   uri: string.isRequired,
   env: string.isRequired,
   feature: string.isRequired,
-  onTrue: oneOfType([func, shape({})]).isRequired,
-  onFalse: oneOfType([func, shape({})]).isRequired,
+  onTrue: oneOfType([func, shape({}), bool]).isRequired,
+  onFalse: oneOfType([func, shape({}), bool]).isRequired,
```
Comment (Iracinder) [inline src/Laika.js:51]: if you're gonna go ahead and type that maybe it's worth typescriptting it

**Review by Iracinder — COMMENTED** [review]
we don't need npmignore anymore?

Also I'm confused as to why we get more react types dependencies when we are phasing them out for typescript?

File: src/Laika.js (hunk 1 above)
Comment (luxalpa) [reply src/Laika.js:51]: I already added the typescript definitions. I think at the current time it's not worth it turning the entire code base into typescript.

**Comment by luxalpa** [comment]
Yes, as you can see in the link it uses "files" instead, which is a whitelist instead of a blacklist. I tested it locally with yalc. It's more secure and easier to use (for example I had the problem that I would need to add my local `.idea` folder to the npmignore list otherwise, even though it's already in my global gitignore. I almost forgot about it).

About the type dependencies I am not sure which ones you are referring to. The typescript typings are added so that it's easier to use in Typescript project; The prop-types are used so that it checks them at runtime. I don't know if we really need those prop-types, but if we do then it should be the correct ones. :) Because the prior version throws an error in the other project.

**Review by Iracinder — APPROVED** [review]

**Pull request merged**